src/apps/tenderize/ethereum/tenderize.tender-tokens.token-fetcher.ts

```diff
@@ -0,0 +1,85 @@
+import { Inject } from '@nestjs/common';
+import Axios from 'axios';
+
+import { IAppToolkit, APP_TOOLKIT } from '~app-toolkit/app-toolkit.interface';
+import { Register } from '~app-toolkit/decorators';
+import { buildDollarDisplayItem } from '~app-toolkit/helpers/presentation/display-item.present';
+import { ContractType } from '~position/contract.interface';
+import { PositionFetcher } from '~position/position-fetcher.interface';
+import { AppTokenPosition } from '~position/position.interface';
+import { Network } from '~types/network.interface';
+
+import { TenderizeContractFactory } from '../contracts';
+import { ethereumEndpoint } from '../helpers/constants';
+import { tenderTokenFetcherQuery, TenderTokenFetcherResponse } from '../helpers/queries';
+import { APYResponse } from '../helpers/types';
+import { TENDERIZE_DEFINITION } from '../tenderize.definition';
+
+const appId = TENDERIZE_DEFINITION.id;
+const groupId = TENDERIZE_DEFINITION.groups.tendertokens.id;
+const network = Network.ETHEREUM_MAINNET;
+
+@Register.TokenPositionFetcher({ appId, groupId, network })
+export class EthereumTenderizeTenderTokensTokenFetcher implements PositionFetcher<AppTokenPosition> {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(TenderizeContractFactory) private readonly tenderizeContractFactory: TenderizeContractFactory,
+  ) {}
+
+  async getPosition(address: string, steak: string, virtualPrice: string, apy: string) {
+    const multicall = this.appToolkit.getMulticall(network);
+    const contract = this.tenderizeContractFactory.erc20({ address, network });
+    const baseTokens = await this.appToolkit.getBaseTokenPrices(network);
+    const underlyingToken = baseTokens.find(v => v.address === steak)!;
+
+    const [symbol, decimals, totalSupply] = await Promise.all([
+      multicall.wrap(contract).symbol(),
+      multicall.wrap(contract).decimals(),
+      multicall.wrap(contract).totalSupply(),
+    ]);
+    const supply = Number(totalSupply) / 10 ** decimals;
+    const price = underlyingToken.price * (Number(virtualPrice) / 10 ** decimals);
+    const token: AppTokenPosition = {
+      address,
+      network,
+      appId,
+      groupId,
+      symbol,
+      decimals,
+      supply,
+      tokens: [],
+      dataProps: {},
+      pricePerShare: 1,
+      price,
+      type: ContractType.APP_TOKEN,
+      displayProps: {
+        label: symbol,
+        secondaryLabel: buildDollarDisplayItem(price),
+        tertiaryLabel: `${apy}% APY`,
+        images: [`https://app.tenderize.me/tender${symbol}.svg`],
```

Comment:
`images: getImagesFromToken(underlyingToken)`

This will use the image of the underlying token. In our UI, we automatically render the Tender logo at the top right of that logo as well, so it'll look nice trust me :)